digdag-core/src/main/java/io/digdag/core/database/migrate/Migration_20190318175338_AddIndexToSessionAttempts.java

```diff
@@ -0,0 +1,23 @@
+package io.digdag.core.database.migrate;
+
+import org.skife.jdbi.v2.Handle;
+
+public class Migration_20190318175338_AddIndexToSessionAttempts
+        implements Migration
+{
+    @Override
+    public void migrate(Handle handle, MigrationContext context)
+    {
+        // DatabaseSessionStoreManager.getActiveAttemptCount uses this index.
+        if (context.isPostgres()) {
+            handle.update("create index concurrently session_attempts_on_site_id_and_state_flags_partial_2 on session_attempts"
+                    + " using btree(site_id) where state_flags & 2 = 0");
```

Comment:
Can the update method wait for the sql completion? I just wanted to confirm it because it'd better that each migration is executed sequentially.

Reply:
Yes. _create index concurrently_ also wait for the completion of sql.
Following is my experiment on above query to about 10,000,000 records.

```
digdag_test=# create index session_attempts_on_site_id_and_state_flags_partial_2 on session_attempts  using btree(site_id) where state_flag
s & 2 = 0;
CREATE INDEX
Time: 2458.866 ms (00:02.459)

digdag_test=# create index concurrently session_attempts_on_site_id_and_state_flags_partial_2 on session_attempts  using btree(site_id) whe
re state_flags & 2 = 0;
CREATE INDEX
Time: 3830.146 ms (00:03.830)
```
With _concurrently_, the response speed is late. So it waited for the completion.

Reply:
Thanks! it makes sense 👍